segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -975,8 +975,9 @@ private Void handleException(long requestId, String segment, long offset, String
             invokeSafely(connection::send, new WireCommands.TableKeyBadVersion(requestId, segment, clientReplyStackTrace), failureHandler);
         } else {
             logError(requestId, segment, operation, u);
-            connection.close(); // Closing connection should reinitialize things, and hopefully fix the problem
-            throw new IllegalStateException("Unknown exception.", u);
+            invokeSafely(connection::send,
```

Comment:
Here you are changing the behavior for this class. You should make sure that `ErrorMessage` only captures those exceptions that it can properly send a message back for, and let the rest fall down to this default behavior. We have a lot of code (on the other side of the Wire Protocol) that relies on these connections to be closed when something unexpected happens.

It is true that we will eventually want to "fix" that code to understand proper error codes, but that is not a task I believe you are willing to undertake now, are you? It's OK to file  follow up issue (not necessarily for yourself) to look at this new Wire Command and make any proper changes (like handling specific errors, etc.) - we can do that later.

Reply:
Sure thing, thanks for pointing that out. I'll create an issue for addressing this.